openhands/server/listen.py

```diff
@@ -308,6 +302,17 @@ async def websocket_endpoint(websocket: WebSocket):
         {"action": "finish", "args": {}}
         ```
     """
+    cookies = dict(
+        cookie.split('=')
+        for cookie in websocket.headers.get('cookie', '').split('; ')
```

Comment:
Did you confirm this works? Not sure that WebSocket's have HTTP headers

Reply:
And yeah--tested both the SaaS and OSS paths here

Reply:
https://stackoverflow.com/a/77060459 Watch out, apparently cookies through WS are not subject to same-origin policy and may be risky